### ISSUE TITLE
[release] Cannot create release on github

### ISSUE THREAD

**Issue by HomeOfTheWizard** [jreleaser/jreleaser#1696]
### Issue 
I am trying to create a release of a java project using the jreleaser maven plugin.
Everything works fine until it tries to create a release on github.
I verified my token by calling the github APIs via curl command, and I do not have any issue creating a release manually.
But when I try with jreleaser, I have a "permenantly moved" 307: Temporary Redirect.

### Task List

- Created a github token and tested.
```
ozgun@ozgun-GL553VD:~$ curl -L   -X POST   -H "Accept: application/vnd.github+json"   -H "Authorization: Bearer XXXXXX"   -H "X-GitHub-Api-Version: 2022-11-28"   https://api.github.com/repos/homeofthewizard/spring-bridge-maven/releases   -d '{"tag_name":"v1.0.0","target_commitish":"main","name":"v1.0.0","body":"Description of the release","draft":false,"prerelease":false,"generate_release_notes":false}'
{
  "url": "https://api.github.com/repos/HomeOfTheWizard/spring-bridge-maven/releases/165347793",
  "assets_url": "https://api.github.com/repos/HomeOfTheWizard/spring-bridge-maven/releases/165347793/assets",
  "upload_url": "https://uploads.github.com/repos/HomeOfTheWizard/spring-bridge-maven/releases/165347793/assets{?name,label}",
  "html_url": "https://github.com/HomeOfTheWizard/spring-bridge-maven/releases/tag/v1.0.0",
  "id": 165347793,
  "author": {
    "login": "HomeOfTheWizard",
    "id": 38886889,
    "node_id": "MDQ6VXNlcjM4ODg2ODg5",
    "avatar_url": "https://avatars.githubusercontent.com/u/38886889?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/HomeOfTheWizard",
    "html_url": "https://github.com/HomeOfTheWizard",
    "followers_url": "https://api.github.com/users/HomeOfTheWizard/followers",
    "following_url": "https://api.github.com/users/HomeOfTheWizard/following{/other_user}",
    "gists_url": "https://api.github.com/users/HomeOfTheWizard/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/HomeOfTheWizard/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/HomeOfTheWizard/subscriptions",
    "organizations_url": "https://api.github.com/users/HomeOfTheWizard/orgs",
    "repos_url": "https://api.github.com/users/HomeOfTheWizard/repos",
    "events_url": "https://api.github.com/users/HomeOfTheWizard/events{/privacy}",
    "received_events_url": "https://api.github.com/users/HomeOfTheWizard/received_events",
    "type": "User",
    "site_admin": false
  },
  "node_id": "RE_kwDOKBn4IM4J2wHR",
  "tag_name": "v1.0.0",
  "target_commitish": "main",
  "name": "v1.0.0",
  "draft": false,
  "prerelease": false,
  "created_at": "2024-07-14T12:10:45Z",
  "published_at": "2024-07-14T12:19:57Z",
  "assets": [

  ],
  "tarball_url": "https://api.github.com/repos/HomeOfTheWizard/spring-bridge-maven/tarball/v1.0.0",
  "zipball_url": "https://api.github.com/repos/HomeOfTheWizard/spring-bridge-maven/zipball/v1.0.0",
  "body": "Description of the release"
```

- followed exactly the maven pom.xml configuration provided here
https://jreleaser.org/guide/latest/examples/maven/maven-central.html
- resulting full Stacktrace 
```
[INFO] --- deploy:3.1.1:deploy (default-deploy) @ spring-bridge-maven ---
[INFO] Using alternate deployment repository local::file:./target/staging-deploy
Uploading to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha.jar
Uploaded to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha.jar (10 kB at 1.3 MB/s)
Uploading to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-javadoc.jar
Uploading to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-sources.jar
Uploaded to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-sources.jar (5.8 kB at 1.9 MB/s)
Uploading to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-build.pom
Uploading to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha.pom
Uploaded to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-build.pom (4.8 kB at 602 kB/s)
Uploaded to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-javadoc.jar (81 kB at 5.4 MB/s)
Uploaded to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha.pom (263 kB at 11 MB/s)
Downloading from local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/maven-metadata.xml
Downloaded from local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/maven-metadata.xml (549 B at 78 kB/s)
Uploading to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/maven-metadata.xml
Uploaded to local: file:./target/staging-deploy/com/homeofthewizard/spring-bridge-maven/maven-metadata.xml (549 B at 275 kB/s)
[INFO] 
[INFO] --- jreleaser:1.13.1:full-release (default-cli) @ spring-bridge-maven ---
[INFO] JReleaser 1.13.1
[INFO]   - basedir set to /home/ozgun/IdeaProjects/spring-bridge-maven-extension
[INFO]   - outputdir set to /home/ozgun/IdeaProjects/spring-bridge-maven-extension/library/target/jreleaser
[INFO] git-root-search set to false
[INFO] Loading variables from /home/ozgun/.jreleaser/config.toml
[INFO] Validating configuration
[INFO] Strict mode set to false
[INFO] Project version set to 1.0.1-alpha
[INFO] Release is not snapshot
[INFO] Timestamp is 2024-07-14T15:41:45.062655873+02:00
[INFO] HEAD is at e7d477d
[INFO] Platform is linux-x86_64
[INFO] dry-run set to false
[INFO] Generating changelog
[INFO] Storing changelog: library/target/jreleaser/release/CHANGELOG.md
[INFO] Calculating checksums for distributions and files
[INFO]   [checksum] No files configured for checksum. Skipping
[INFO] Cataloging artifacts
[INFO]   Cataloging is not enabled. Skipping
[INFO] Signing distributions and files
[INFO]   [sign] No files configured for signing. Skipping
[INFO] Deploying Maven artifacts
[INFO]   [maven] Deploying all staged artifacts
[INFO]     [mavenCentral] Deploying to sonatype
[INFO]     [mavenCentral] Verifying prerequesites
[INFO]     [mavenCentral] Verifying POMs
[INFO]     [mavenCentral] Checking if key 2CAA79415021BA48 has been published
[INFO]     [mavenCentral] Key 2CAA79415021BA48 was found as published
[INFO]     [sign] target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-sources.jar
[INFO]     [sign] target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha.pom
[INFO]     [sign] target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-build.pom
[INFO]     [sign] target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha-javadoc.jar
[INFO]     [sign] target/staging-deploy/com/homeofthewizard/spring-bridge-maven/1.0.1-alpha/spring-bridge-maven-1.0.1-alpha.jar
[INFO]     [mavenCentral]  - com.homeofthewizard-spring-bridge-maven-1.0.1-alpha-bundle.zip
[INFO]     [mavenCentral] Attempt 1 of 61
[INFO] Writing output properties to library/target/jreleaser/output.properties
[ERROR] JReleaser failed after 46.264 s
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  01:03 min
[INFO] Finished at: 2024-07-14T15:42:32+02:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.13.1:full-release (default-cli) on project spring-bridge-maven: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.13.1:full-release failed: Unexpected error: deployment 3b97ae89-f7eb-4aef-b195-d293a5976527 failed
[ERROR] pkg:maven/com.homeofthewizard/guice-exporter-maven-extension@1.0.1-alpha Component with package url: 'pkg:maven/com.homeofthewizard/guice-exporter-maven-extension@1.0.1-alpha' already exists
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the '-e' switch
[ERROR] Re-run Maven using the '-X' switch to enable verbose output
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```
- link to git repository 
https://github.com/HomeOfTheWizard/spring-bridge-maven

### Steps to Reproduce
run the following command to create a release.
`./mvnw -pl library -Ppublication deploy jreleaser:full-release

### Expected Behaviour
create a release package, deploy to maven central, create a release on github.

### Actual Behaviour
uploads to maven central, but fails when creating a release on github.

upload on maven central: https://central.sonatype.com/artifact/com.homeofthewizard/guice-exporter-maven-extension/1.0.1-alpha

but fails creating the release:
```
[INFO] Uploading distributions and files
[INFO]   [upload] Uploading is not enabled. Skipping
[INFO] Releasing to https://github.com/HomeOfTheWizard/spring-bridge-maven-extension@main
[INFO] Writing output properties to extension/target/jreleaser/output.properties
[ERROR] JReleaser failed after 09:42 m
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  09:50 min
[INFO] Finished at: 2024-07-14T15:27:19+02:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.13.1:full-release (default-cli) on project guice-exporter-maven-extension: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.13.1:full-release failed: Unexpected error when creating release: org.jreleaser.sdk.commons.RestAPIException: 307: Temporary Redirect
[ERROR] {"message":"Moved Permanently","url":"https://api.github.com/repositories/672790560/releases","documentation_url":"https://docs.github.com/rest/guides/best-practices-for-using-the-rest-api#follow-redirects"}
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the '-e' switch
[ERROR] Re-run Maven using the '-X' switch to enable verbose output
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

### Environment Information

- **Operating System**: ubuntu 24.4
- **JReleaser Version**: maven plugin v1.13.1
- **JDK Version:** openjdk 17

**Comment by HomeOfTheWizard** [comment]
[trace.log](https://github.com/user-attachments/files/16227660/trace.log)

**Comment by aalmiray** [comment]
Looks like the GH API expects the client to follow a redirect but the client can not handle that option at the moment. This is definitely a bug.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.14.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.14.0))
